### PR TITLE
Fixed: Visual states not working properly when the animation is canceled

### PR DESCRIPTION
**Pull request by tzraikov**
#1905

### CODE REVIEW

File: ui/animation/animation.ios.ts [hunk 1]
```diff
@@ -144,9 +144,10 @@ export class Animation extends common.Animation implements definition.Animation
         let i = 0;
         let length = this._mergedPropertyAnimations.length;
         for (; i < length; i++) {
-            (<UIView>this._mergedPropertyAnimations[i].target._nativeView).layer.removeAllAnimations();
-            if ((<any>this._mergedPropertyAnimations[i])._propertyResetCallback) {
-               (<any>this._mergedPropertyAnimations[i])._propertyResetCallback((<any>this._mergedPropertyAnimations[i])._originalValue);
+            let propertyAnimation = this._mergedPropertyAnimations[i];
+            propertyAnimation.target._nativeView.layer.removeAllAnimations();
+            if ((<any>propertyAnimation)._propertyResetCallback) {
```
Comment (hshristov) [inline ui/animation/animation.ios.ts]: why do we cast to `any`?

**Comment by PanayotCankov** [comment]
:+1:

File: ui/animation/keyframe-animation.ts [hunk 2]
```diff
@@ -98,6 +100,16 @@ export class KeyframeAnimation {
...
+            if (this._currentAnimation && this._currentAnimation.isPlaying) {
+                this._currentAnimation.cancel();
+            }
+            this._isPlaying = false;
+            this._rejectAnimationFinishedPromise();
+        }
+    }
+
     public play(view: view.View): Promise<void> {
         if (this._isPlaying) {
```
Comment (hshristov) [inline ui/animation/keyframe-animation.ts]: Do we need to throw exception here? Can we just return if running?

**Comment by hshristov** [comment]
:+1:

**Pull request merged**